.github/workflows/checks.yml

```diff
@@ -187,3 +187,43 @@ jobs:
     - name: Run pytest
       run: |
         pytest --pyargs watertap
+
+  macos:
+    name: macOS setup (EXPERIMENTAL)
+    runs-on: macos-10.15
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - '3.8'
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        activate-environment: watertap
+        python-version: ${{ matrix.python-version }}
+    - name: Install WaterTAP (dev) without idaes get-extensions
+      run: |
+        echo '::group::Output of "conda install" commands'
+        conda install --quiet --yes pip=21.1 wheel setuptools pandoc
+        echo '::endgroup::'
+        echo '::group::Output of "pip install" commands'
+        pip install --progress-bar off -r requirements-dev.txt
+        echo '::endgroup::'
+        echo '::group::Display installed packages'
+        conda list
+        pip list
+        pip show pyomo idaes-pse
+        echo '::endgroup::'
+    - name: Install Ipopt from conda-forge
+      run:
+        conda install --quiet --yes -c conda-forge ipopt
+    - name: Build Pyomo extensions
+      run: |
+        conda install --quiet --yes cmake
+        # some failures are expected, but this should succeed as long as pynumero is built correctly
+        pyomo build-extensions || python -c "from pyomo.contrib.pynumero.asl import AmplInterface; exit(0) if AmplInterface.available() else exit(1)"
```

Comment:
Nice - that's a much better way to check if `pyomo build-extensions` is successful than trying to parse its output.

Reply:
For some reason pipe was still failing, even though it worked on my system. GHA might be checking the output of each command when pipe is used.

At any rate, I do like the direct import checking better as well.